sensor/kubernetes/sensor/sensor.go

```diff
@@ -106,7 +106,7 @@ func CreateSensor(cfg *CreateOptions) (*sensor.Sensor, error) {
 
 	imageCache := expiringcache.NewExpiringCache(env.ReprocessInterval.DurationSetting())
 	policyDetector := detector.New(enforcer, admCtrlSettingsMgr, resources.DeploymentStoreSingleton(), resources.ServiceAccountStoreSingleton(), imageCache, auditLogEventsInput, auditLogCollectionManager, resources.NetworkPolicySingleton())
-	pipeline := eventpipeline.New(cfg.k8sClient, configHandler, policyDetector, k8sNodeName.Setting(), cfg.resyncPeriod, cfg.traceWriter, storeProvider)
+	pipeline := eventpipeline.New(storeProvider.Nodes(), cfg.k8sClient, configHandler, policyDetector, k8sNodeName.Setting(), cfg.resyncPeriod, cfg.traceWriter, storeProvider)
```

Comment:
I don't think you need to pass the `NodeStore` here. We are already passing the `storeProvider` that contains the `NodeStore`. You can just use `storeProvider.Nodes()` when you are creating the `nodeDispatcher` in the `NewDispatcherRegistry` function.

Reply:
You are absolutely right. I got that corrected in [660a42c](https://github.com/stackrox/stackrox/pull/4484/commits/660a42c645c6912ae3bb6596d3645c764684dba7).